test/integration/admin/dashboard/dashboard.js

```diff
@@ -0,0 +1,56 @@
+context('dashboard', function() {
+  specify('display dashboard', function() {
+    cy
+      .server()
+      .routeDashboards()
+      .routeDashboard(fx => {
+        fx.data.id = '1';
+        fx.data.attributes.name = 'Test Dashboard';
+        fx.data.attributes.embed_url = '/test_dashboard';
+
+        return fx;
+      })
+      .visit('/dashboards/1')
+      .wait('@routeDashboard');
+
+    cy
+      .get('.dashboard__frame')
+      .find('.dashboard__context-trail')
+      .should('contain', 'Test Dashboard');
+
+    cy
+      .get('.dashboard__frame')
+      .find('.dashboard__iframe iframe')
+      .should('have.attr', 'src', '/test_dashboard');
+
+    cy
+      .get('.dashboard__frame')
+      .find('.dashboard__context-trail .js-back')
+      .click();
+
+    cy
+      .url()
+      .should('not.contain', 'dashboards/1')
+      .should('contain', 'dashboards');
+  });
+
+  specify('dashboard does not exist', function() {
```

Comment:
I don't understand why this test works. how does returning null fail?  I think this should be testing for if the route returns a 404 or something.. but I suspect this is "working" because of a javascript error of the data returning something in an unexpected format which should really never happen.

Reply:
Yeah... not sure what I was thinking there.